lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -0,0 +1,107 @@
+require 'fastlane/action'
+require 'digest/sha1'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      S3_UPLOADED_FILE_PATH = :S3_UPLOADED_FILE_PATH
+    end
+
+    class UploadToS3Action < Action
+      def self.run(params)
+        file_path = params[:file]
+        file_name = File.basename(file_path)
+        UI.user_error!("Unable to read file at #{file_path}") unless File.file?(file_path)
+
+        bucket = params[:bucket]
+        key = params[:key]
+
+        if params[:auto_prefix] == true
+          file_name_hash = Digest::SHA1.base64digest(file_name)
+          key = [file_name_hash, key].join('/')
```

Comment:
Just curious. Is there an advantage of doing this over `"#{file_name_hash}/#{key}"?

Reply:
Ahah I had the exact same question.

Actually during my first review my skimming gave me the wrong impression that the `params[:key]` provided by the call site would be unused if `auto_prefix` was true, because this line would overwrite its value with `key = …`. Not sure if using `"#{file_name_hash}/#{key}"` would have prevented my skimming to have the same wrong assumption (lack of coffee was probably the real culprit) but I was still caught off-guard by this 😆 

---

I'm not fully an S3 expert, so I have to ask: what's the result of having a `/` in that prefixed `key`? Is this still considered to be a single key name (with literal character`/` in it) in S3 API, or will that result into an actual intermediate directory in the path to the file? I believe it's the latter, right? If so, then maybe even using `File.join` would make even more sense here then?

```suggestion
          key = File.join(file_name_hash, key)
```

Reply:
I initially used `File.join`, then realized that the result would be invalid on Windows, where the path separator is `\`. 

I'm not sure we're likely to have this running on windows boxes _soon_, but I figured this was safe since it's S3-specific 🙂

Reply:
Good point 👍